examples/ensemble/plot_random_forest_regression_criteria_comparison.py

```diff
@@ -133,50 +166,60 @@ def main(simulation_name, n_samples, criterion, n_dimensions, n_iter):
 n_dimensions = 10
 simulation_names = simulations.keys()
 sample_sizes = np.arange(5, 51, 3)
-criteria = ["mae", "mse", "friedman_mse", "axis", "oblique"]
+criteria = ["mae", "mse", "friedman_mse"]
 
 # Number of times to repeat each simulation setting
-n_repeats = 10
+n_repeats = 30
 
 # Create the parameter space
-params = product(simulation_names, sample_sizes, criteria,
-                 [n_dimensions], range(n_repeats))
+params = product(simulation_names, sample_sizes, criteria, range(n_repeats))
 
 
 ###############################################################################
 print("Constructing validation datasets...")
 for simulation_name, (sim, noise) in simulations.items():
     if noise is not None:
-        X_test, y_test = sim(n_samples=1000,
-                             n_dimensions=n_dimensions,
-                             noise=noise,
-                             random_state=random_state)
+        X_test, y_test = sim(
+            n_samples=1000,
+            n_dimensions=n_dimensions,
+            noise=noise,
+            random_state=random_state,
+        )
     else:
-        X_test, y_test = sim(n_samples=1000,
-                             n_dimensions=n_dimensions,
-                             random_state=random_state)
+        X_test, y_test = sim(
+            n_samples=1000, n_dimensions=n_dimensions, random_state=random_state
+        )
     simulations[simulation_name].append((X_test, y_test))
 
 
 ###############################################################################
 print("Running simulations...")
 
-with Pool() as pool:
-
-    # Run the simulations in parallel
-    data = pool.starmap(main, params)
-
-    # Save results as a DataFrame
-    columns = ["simulation", "n_samples", "criterion",
-               "n_dimensions", "mse", "runtime"]
-    df = pd.DataFrame(data, columns=columns)
-
-    # Plot the results
-    sns.relplot(x="n_samples",
-                y="mse",
-                hue="criterion",
-                col="simulation",
-                kind="line",
-                data=df,
-                facet_kws={'sharey': False, 'sharex': True})
-    plt.show()
+# Generate training and test data for simulations
+sim_data = {}
+for sim in simulation_names:
+    sim_data = _prep_data(sim_data, sim, sample_sizes[-1], n_dimensions, n_repeats)
+
+# Run the simulations in parallel
+data = Parallel(n_jobs=-2)(
+    delayed(main)(sim_name, sim_data[sim_name][n_iter], n, crit, n_dimensions, n_iter)
+    for sim_name, n, crit, n_iter in params
+)
+
+# Save results as a DataFrame
+columns = ["simulation", "n_samples", "criterion", "n_dimensions", "mse", "runtime"]
+df = pd.DataFrame(data, columns=columns)
+
+# Plot the results
+sns.relplot(
+    x="n_samples",
+    y="mse",
+    hue="criterion",
+    col="simulation",
+    kind="line",
+    data=df,
+    facet_kws={"sharey": False, "sharex": True},
+)
```

Comment:
Not sure if `scikit-learn` will like this because I don't believe `pandas` or `seaborn` are part of the default installation... but let's keep it in since it makes plotting so much easier!